table/block_based/block_based_table_reader.cc

```diff
@@ -1172,9 +1193,10 @@ Status BlockBasedTable::GetDataBlockFromCache(
         read_options.fill_cache) {
       size_t charge = block_holder->ApproximateMemoryUsage();
       Cache::Handle* cache_handle = nullptr;
-      auto deleter = BlocklikeTraits<TBlocklike>::GetDeleter(block_type);
-      s = block_cache->Insert(block_cache_key, block_holder.get(), charge,
-                              deleter, &cache_handle);
+      s = block_cache->Insert(
+          block_cache_key, block_holder.get(),
+          BlocklikeTraits<TBlocklike>::GetCacheItemHelper(block_type), charge,
+          &cache_handle, priority);
```

Comment:
Interesting. This is actually an improvement from before. Previously, any block copied from the compressed to the uncompressed cache was low priority.